pkg/apis/postgres-operator.crunchydata.com/v1beta1/standalone_pgadmin_types.go

```diff
@@ -97,6 +97,30 @@ type PGAdminSpec struct {
 	// More info: https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration
 	// +optional
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// AdminUsername is the username to set during pgAdmin startup
```

Comment:
Adding comment. I had this as "AdminEmail" at one point but people aren't actually putting in a real email, just something with a valid email form. Do you have a preference on naming this?

Reply:
What happens if this field is in the wrong format?

This feels like something we should try to prevent with validation. We _should not_ implement an email regex here. OpenAPI [may have a `format=email`](https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-00#section-7.3), but I don't know that Kubernetes recognizes it.

Reply:
> What happens if this field is in the wrong format?

This value is used in the [setup/database migration](https://github.com/pgadmin-org/pgadmin4/blob/REL-7_7/web/migrations/versions/fdc58d9bd449_.py#L38-L139) to create the first/administrator user. Since the name/pw are env vars, they are [not subject to validation](https://github.com/pgadmin-org/pgadmin4/blob/REL-7_7/web/pgadmin/setup/user_info.py#L37-L44).

So if the username isn't a valid email, pgadmin starts up just fine.

But InternalAuthentication does check for a [valid email](https://github.com/pgadmin-org/pgadmin4/blob/REL-7_7/web/pgadmin/utils/validation_utils.py#L10-L19), so if we want this user to be able to log in, it needs to be in the right format. (pgadmin4 v4 had a different [validation method](https://github.com/pgadmin-org/pgadmin4/blob/REL-4_30/web/pgadmin/utils/validation_utils.py) which we could get around with just a `@pgo` suffix.)

Reply:
> This feels like something we should try to prevent with validation

Yeah. It looks like `format=email` is not defined. (Though there is an `email` field under [`ContactObject`](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md#fixed-fields-2)?)

`+kubebuilder:validation:Pattern` might do it, but that's an email regex.

Hmmm....

Reply:
Thank you for all those links.

It looks like it's the "globally deliverable" aspect of the validator that cares. The "check deliverability" option also caught my eye. The pgAdmin default seems to be `False` for now.

There's nothing more I need here. The field comment is sufficient, IMO.

Reply:
I've got a section in my (nascent) docs for discussing this field, so I'll put a caution in there too about this: your pgadmin will start up, but you won't be able to log in.